### DISCUSSION TITLE
Read sensor only at certain hour

### DISCUSSION THREAD

**Discussion by Sill1000** [HomeSpan/HomeSpan#596] [Q&A] [answered]
Hello

How to read the sensor only over a specific time slot (for ex. from 8h00 until 22h00)
It seems that the NTP time is used for the Weblog. 
How to get the time for using in the code ?

Thanks

**Comment by HomeSpan** [comment, marked as answer]
Once the WebLog has acquired the time, you can use any standard C++ time function to retrieve, format, print, etc., the current date and time.  The Arduino-ESP32 also has a useful helper function, `getLocalTime()`, that would likely work well for your use case.  Here is an example of how it is used:

```C++
    struct tm cTime;   // first, create a standard C time structure
    getLocalTime(&cTime);    // next, populate the structure with the current date and time
    Serial.printf("Current Time = %02d:%02d:%02d\n",cTime.tm_hour,cTime.tm_min,cTime.tm_sec);
```

For details, see for example: https://techtutorialsx.com/2021/09/01/esp32-system-time-and-sntp/

**Reply by Sill1000** [reply]
Perfect! Thanks !

**Comment by TheDevNino** [comment]
@Sill1000 could you show how you've implemented it into your code?
